src/cider/exec/nextgen/operators/HashJoinNode.cpp

```diff
@@ -0,0 +1,254 @@
+    } else {
+      // null buffer decoder
+      // TBD: Null representation, bit-array or bool-array.
+      auto row_null_data = func.emitRuntimeFunctionCall(
+          "check_bit_vector_clear",
+          JITFunctionEmitDescriptor{
+              .ret_type = JITTypeTag::BOOL,
+              .params_vector = {{varsize_values.getNull().get(), index_.get()}}});
+      expr_->set_expr_value(row_null_data, len, row_data);
+    }
+  }
+
+  void readFixSizedTypeCol() {
+    utils::FixSizeJITExprValue fixsize_values(buffer_values_);
+    auto data_buffer = fixsize_values.getValue();
+    auto& func = data_buffer->getParentJITFunction();
+    JITTypeTag tag = utils::getJITTypeTag(expr_->get_type_info().get_type());
+    // data buffer decoder
+    auto actual_raw_data_buffer = data_buffer->castPointerSubType(tag);
+    auto row_data = getFixSizeRowData(func, fixsize_values);
+    if (expr_->get_type_info().get_notnull()) {
+      expr_->set_expr_value(func.createLiteral(JITTypeTag::BOOL, false), row_data);
+    } else {
+      // null buffer decoder
+      // TBD: Null representation, bit-array or bool-array.
+      auto row_null_data = func.emitRuntimeFunctionCall(
+          "check_bit_vector_clear",
+          JITFunctionEmitDescriptor{
+              .ret_type = JITTypeTag::BOOL,
+              .params_vector = {{fixsize_values.getNull().get(), index_.get()}}});
+
+      expr_->set_expr_value(row_null_data, row_data);
+    }
+  }
+
+  JITValuePointer getFixSizeRowData(JITFunction& func,
+                                    utils::FixSizeJITExprValue& fixsize_val) {
+    if (expr_->get_type_info().get_type() == kBOOLEAN) {
+      auto row_data = func.emitRuntimeFunctionCall(
+          "check_bit_vector_set",
+          JITFunctionEmitDescriptor{
+              .ret_type = JITTypeTag::BOOL,
+              .params_vector = {{fixsize_val.getValue().get(), index_.get()}}});
+      return row_data;
+    } else {
+      JITTypeTag tag = utils::getJITTypeTag(expr_->get_type_info().get_type());
+      // data buffer decoder
+      auto data_pointer = fixsize_val.getValue()->castPointerSubType(tag);
+      auto row_data = data_pointer[index_];
+      return row_data;
+    }
+  }
+
+ private:
+  utils::JITExprValue& buffer_values_;
+  ExprPtr& expr_;
+  JITValuePointer& index_;
+};
+
+TranslatorPtr HashJoinNode::toTranslator(const TranslatorPtr& succ) {
+  return createOpTranslator<HashJoinTranslator>(shared_from_this(), succ);
+}
+
+void HashJoinTranslator::consume(context::CodegenContext& context) {
+  codegen(context);
+}
+
+// traverse join_quals expr tree to get join key value and null vector
+void traverse(ExprPtr expr,
+              jitlib::JITFunctionPointer func,
+              std::vector<JITValuePointer>& keys,
+              std::vector<JITValuePointer>& nulls,
+              context::CodegenContext& context) {
+  if (Analyzer::ColumnVar* col_var = dynamic_cast<Analyzer::ColumnVar*>(expr.get())) {
+    if (col_var->get_table_id() == 100) {
```

Comment:
I think you'd better add a FIXME here since `100` is not always used as right table id.

Reply:
FIX